### PR TITLE
feat(twitch): Add integrations for Twitch patches

### PR DESCRIPTION
**Pull request by ThePBone**
Used by:
* https://github.com/revanced/revanced-patches/pull/1074
* https://github.com/revanced/revanced-patches/pull/1075

### CODE REVIEW

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]
This PR adds quite a lot of boilerplate duplicate code. If possible reduce it.

**Comment by ThePBone** [comment]
I cleaned it up a bit and removed some unnecessary code. There is some overlap with the utility classes from YouTube/Tiktok integrations, but I didn't import their classes to prevent confusion.

File: app/src/main/java/app/revanced/twitch/settingsmenu/SettingsHooks.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    /**
+     * Intercepts fragment loading in SettingsActivity.onCreate
+     * @return Returns true if the revanced settings have been requested by the user, otherwise false
+     */
+    public static boolean handleSettingsCreation(AppCompatActivity base) {
+        if(!base.getIntent().getBooleanExtra(EXTRA_REVANCED_SETTINGS, false)) {
+            LogHelper.debug("Revanced settings not requested");
+            return false; // User wants to enter another settings fragment
+        }
+        LogHelper.debug("Revanced settings requested");
```
Comment (oSumAtrIX) [inline app/src/main/java/app/revanced/twitch/settingsmenu/SettingsHooks.java:97]:
```suggestion
        LogHelper.debug("ReVanced settings requested");
```

File: app/src/main/java/app/revanced/twitch/settingsmenu/SettingsHooks.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import app.revanced.twitch.utils.LogHelper;
+import tv.twitch.android.feature.settings.menu.SettingsMenuGroup;
+import tv.twitch.android.settings.SettingsActivity;
+
+public class SettingsHooks {
+    private static final int REVANCED_SETTINGS_MENU_ITEM_ID = 0x7;
+    private static final String EXTRA_REVANCED_SETTINGS = "app.revanced.twitch.settings";
+
+    /**
+     * Launches SettingsActivity and show revanced settings
```
Comment (oSumAtrIX) [inline app/src/main/java/app/revanced/twitch/settingsmenu/SettingsHooks.java:25]:
```suggestion
     * Launches SettingsActivity and show ReVanced settings
```

File: app/src/main/java/app/revanced/twitch/settingsmenu/SettingsHooks.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+     */
+    public static void startSettingsActivity() {
+        LogHelper.debug("Launching revanced settings");
```
Comment (oSumAtrIX) [inline app/src/main/java/app/revanced/twitch/settingsmenu/SettingsHooks.java:28]:
```suggestion
        LogHelper.debug("Launching ReVanced settings");
```

File: app/src/main/java/app/revanced/twitch/settingsmenu/SettingsHooks.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+            c.startActivity(intent);
+        });
+    }
+
+    /**
+     * Helper for easy access in smali
+     * @return Returns string resource id
+     */
+    public static int getRevancedSettingsString() {
```
Comment (oSumAtrIX) [inline app/src/main/java/app/revanced/twitch/settingsmenu/SettingsHooks.java:44]: Probably want to use the casing of `ReVanced` here.

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

**Review by oSumAtrIX — APPROVED** [review]

File: app/src/main/java/app/revanced/twitch/settingsmenu/SettingsHooks.java (hunk 4 above)
Comment (ThePBone) [reply app/src/main/java/app/revanced/twitch/settingsmenu/SettingsHooks.java]: Fixed with 2665f5a

**Pull request merged**